src/main/java/core/mate/academy/service/TruckProducer.java

```diff
@@ -0,0 +1,17 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TruckProducer implements MachineProducer<Machine> {
```

Comment:
Parameterize `MachineProducer` with `Truck` to prevent the creation of a producer that does not use `Machine` or its subtype. The current implementation allows for the creation of producers that do not comply with this restriction.